drivers/ad5770r/ad5770r.h

```diff
@@ -0,0 +1,377 @@
+	bool					alarm_config[8];
+	enum ad5770r_output_filter_resistor	output_filter[6];
+	struct monitor_setup			mon_setup;
+	bool					mask_hw_ldac[6];
+	uint16_t				dac_value[6];
+	struct dac_page_mask			page_mask;
+	bool					mask_channel_sel[6];
+	bool 					sw_ldac[6];
+	uint16_t				input_value[6];
+	bool					channel_enable[6];
+};
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+int32_t ad5770r_spi_reg_read(struct ad5770r_dev *dev,
+			     uint8_t reg_addr,
+			     uint8_t *reg_data);
+int32_t ad5770r_spi_reg_read_multiple(struct ad5770r_dev *dev,
+				      uint8_t reg_addr,
+				      uint8_t *reg_data,
+				      uint16_t count);
+int32_t ad5770r_spi_reg_write(struct ad5770r_dev *dev,
+			      uint8_t reg_addr,
+			      uint8_t reg_data);
+int32_t ad5770r_spi_reg_write_multiple(struct ad5770r_dev *dev,
+				       uint8_t reg_addr,
+				       uint8_t *reg_data,
+				       uint16_t count);
+int32_t ad5770r_spi_write_mask(struct ad5770r_dev *dev,
+			       uint8_t reg_addr,
+			       uint32_t mask,
+			       uint8_t data);
+int32_t ad5770r_set_device_spi(struct ad5770r_dev *dev,
+			       struct device_spi_settings spi_settings);
+int32_t ad5770r_channel_config(struct ad5770r_dev *dev,
+			       bool *channel_config);
+int32_t ad5770r_set_output_mode(struct ad5770r_dev *dev,
+				struct output_range output_mode,
+				enum ad5770r_channels channel);
+int32_t ad5770r_set_reference(struct ad5770r_dev *dev,
+			      bool external_reference,
+			      enum ad5770r_reference_voltage reference_selector);
+int32_t ad5770r_set_alarm(struct ad5770r_dev *dev,
+			  bool * const alarm_config);
+int32_t ad5770r_set_output_filter(struct ad5770r_dev *dev,
+				  enum ad5770r_output_filter_resistor output_filter,
+				  enum ad5770r_channels channel);
+int32_t ad5770r_set_hw_ladc(struct ad5770r_dev *dev,
+			    bool * const mask_hw_ldac);
+int32_t ad5770r_set_dac_value(struct ad5770r_dev *dev,
+			      uint16_t dac_value, enum ad5770r_channels channel);
+int32_t ad5770r_set_dac_input(struct ad5770r_dev *dev,
+			      uint16_t dac_input, enum ad5770r_channels channel);
+int32_t ad5770r_set_page_mask(struct ad5770r_dev *dev,
+			      struct dac_page_mask page_mask);
+int32_t ad5770r_set_mask_channel(struct ad5770r_dev *dev,
+				 bool *mask_channel_sel);
+int32_t ad5770r_set_sw_ldac(struct ad5770r_dev *dev,
+			    bool *sw_ldac);
+int32_t ad5770r_set_channel_en(struct ad5770r_dev *dev,
+			       bool *channel_enable);
+int32_t ad5770r_get_status(struct ad5770r_dev *dev,
+			   uint8_t *status);
+int32_t ad5770r_set_monitor_setup(struct ad5770r_dev *dev,
+				  struct monitor_setup mon_setup);
+int32_t ad5770r_init(struct ad5770r_dev **device,
+		     struct ad5770r_init_param init_param);
+
+#endif // AD5770R_H_0
```

Comment:
nitpick: this looks like a typo;  `AD5770R_H_0` -> `AD5770R_H_`; 
since this line may change, maybe use comment style `/* AD5770R_H_ */`